### PR TITLE
Feature/ddo-validate

### LINKED ISSUE #400: Validate DDO before publish
Validate DDO before publish/update by:

- create a new function in src/metadatacache/MetadataCache.ts which calls aquarius's endpoint validateDDO
- update src/ocean/Assets.ts and alter create & updateMetadata functions to validate a DDO before publish/update
- have unit tests with bad DDO

### PR DESCRIPTION
**Pull request by 0x3bfc**
closes #400

### CODE REVIEW

File: src/metadatacache/MetadataCache.ts [hunk 1]
```diff
@@ -312,6 +312,19 @@ export class MetadataCache {
...
   }
 
+  /**
+   * Validate DDO.
+   * @param  {ddp} string DDO.
+   * @return {Promise<any>} Result (200 OK, 500 INVALID).
+   */
+  public async validate(ddo: DDO): Promise<any> {
+    return await this.fetch
+      .post(`${this.url}${apiPath}/validate`, JSON.stringify(ddo))
```
Comment (ssallam) [inline src/metadatacache/MetadataCache.ts:322]: The validate endpoint expects the metadata json not the whole DDO. Get the metadata service `ddo.service[0].attributes`

Comment (kremalicious) [reply src/metadatacache/MetadataCache.ts:322]: `ddo.findServiceByType('metadata)`. Never assuming the metadata service is the first service is safer.

**Comment by kremalicious** [comment]
My main question would be: what is it validating? json structure? metadata structure? Plecos? This should be mentioned somewhere what exactly this validates.

File: src/metadatacache/MetadataCache.ts (hunk 1 above)
Comment (0x3bfc) [reply src/metadatacache/MetadataCache.ts:322]: thanks for the feedback .. working on it.

File: src/metadatacache/MetadataCache.ts [hunk 2]
```diff
@@ -218,6 +218,61 @@ export class MetadataCache {
...
 
   /**
+   * Edit Metadata for a DDO.
+   * @param  {did} string DID.
+   * @param  {newMetadata}  EditableMetadata Metadata fields & new values.
+   * @param  {String} updated Updated field of the DDO
+   * @param  {String} signature Signature using updated field to verify that the consumer has rights
+   * @return {Promise<String>} Result.
+   */
+  public async editMetadata(
```
Comment (kremalicious) [inline src/metadatacache/MetadataCache.ts:228]: so there has been a drastic change, this is now basically in `ocean.assets.editMetadata()` but it only does an object restructuring https://github.com/oceanprotocol/ocean.js/blob/main/src/ocean/Assets.ts#L253

and then the real update is done with `ocean.assets.updateMetadata()` https://github.com/oceanprotocol/ocean.js/blob/main/src/ocean/Assets.ts#L275

super confusing naming and all but if we want to validate anything it needs to happen over there

Comment (0x3bfc) [reply src/metadatacache/MetadataCache.ts:228]: >> if we want to validate anything it needs to happen over there

You mean in the `ocean.assets.updateMetadata()` ?

**Pull request closed without merging**